.pre-commit-config.yaml

```diff
@@ -1,7 +1,23 @@
-exclude: '^$'
-fail_fast: false
 repos:
 -   repo: https://github.com/python/black
     rev: '19.3b0'
     hooks:
     -   id: black
+-   repo: https://github.com/pre-commit/mirrors-mypy
+    rev: 'v0.701'
+    hooks:
+    -   id: mypy
+        name: mypy-ml-agents
+        files: "ml-agents/.*"
+        args: [--ignore-missing-imports, --disallow-incomplete-defs]
```

Comment:
What are --ignore-missing-imports, --disallow-incomplete-defs used for?

Reply:
`--disallow-incomplete-defs` means that the checker will complain about functions that have some arguments and return annotated, but not others. For example
``` python
def foo(x: int, y) -> int:  # bad, no type hint on y

def bar(x: int, y: int):  # bad, no return type
```
would be treated as errors. I didn't apply it to `ml-agents-env` because there's still some code that I'm not sure what the types should be.

`--ignore-missing-imports` is a little more complicated; without it the checker will complain about importing numpy and tensorflow. It also complains about some imports in the mlagents repo; I think is related to our directory structure (similar to the package problems we had during the release) but I'll admit that I don't fully understand it yet. I would like to remove these in the future but it's the only practical way to get checks passing today.